101-expressroute-private-peering-vnet/azuredeploy.json

```diff
@@ -0,0 +1,329 @@
+      }
+    },
+    "subnet1Name": {
+      "type": "string",
+      "defaultValue": "subnet1",
+      "metadata": {
+        "description": "name of the subnet"
+      }
+    },
+    "vnetAddressSpace": {
+      "type": "string",
+      "metadata": {
+        "description": "address space assigned to the VNet"
+      }
+    },
+    "subnet1Prefix": {
+      "type": "string",
+      "metadata": {
+        "description": "network prefix assigned to the subnet"
+      }
+    },
+    "gatewaySubnetPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "network prefixes assigned to the gateway subnet. it has to be a network prefix with mask /27 or larger"
+      }
+    },
+    "gatewayName": {
+      "type": "string",
+      "metadata": {
+        "description": "name of the ExpressRoute Gateway has to be deployed in the VNet"
+      }
+    },
+    "gatewaySku": {
+      "type": "string",
+      "defaultValue": "HighPerformance",
+      "allowedValues": [
+        "Standard",
+        "HighPerformance",
+        "UltraPerformance",
+        "ErGw1AZ",
+        "ErGw2AZ",
+        "ErGw3AZ"
+      ],
+      "metadata": {
+        "description": "ExpressRoute Gateway SKU"
+      }
+    }
+  },
+  "variables": {
+    "erAzureSubscriptionId": "[parameters('erAzureSubscriptionId')]",
+    "erResourceGroup": "[parameters('erResourceGroup')]",
+    "erlocation": "[parameters('location')]",
+    "erCircuitName": "[parameters('erCircuitName')]",
+    "erCircuitId": "[resourceId(variables('erAzureSubscriptionId'),variables('erResourceGroup'),'Microsoft.Network/expressRouteCircuits',variables('erCircuitName'))]",
+    "erSKU_Tier": "[parameters('erSKU_Tier')]",
+    "erSKU_Family": "[parameters('erSKU_Family')]",
+    "erSKU_Name": "[concat(variables('erSKU_Tier'),'_',variables('erSKU_Family'))]",
+    "peerASN": "[parameters('peerASN')]",
+    "primaryPeerAddressPrefix": "[parameters('primaryPeerAddressPrefix')]",
+    "secondaryPeerAddressPrefix": "[parameters('secondaryPeerAddressPrefix')]",
+    "vlanId": "[parameters('vlanId')]",
+    "location": "[parameters('location')]",
+    "vnetName": "[parameters('vnetName')]",
+    "subnet1Name": "[parameters('subnet1Name')]",
+    "vnetAddressSpace": "[parameters('vnetAddressSpace')]",
+    "subnet1Prefix": "[parameters('subnet1Prefix')]",
+    "gatewaySubnetPrefix": "[parameters('gatewaySubnetPrefix')]",
+    "gatewayName": "[parameters('gatewayName')]",
+    "gatewayPublicIPName": "[concat(variables('gatewayName'),'-pubIP')]",
+    "gatewaySku": "[parameters('gatewaySku')]",
+    "gatewaySubnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('vnetName'), 'GatewaySubnet')]",
+    "nsg": "nsg"
+  },
+  "resources": [
+    {
+      "apiVersion": "2019-12-01",
+      "type": "Microsoft.Network/expressRouteCircuits",
+      "name": "[variables('erCircuitName')]",
+      "location": "[variables('erlocation')]",
+      "comments": "Create the ER circuit",
```

Comment:
move comment properties to the top of the resource definition (see contribution guide for sort order)